### PR TITLE
[explorer] validator map followup

### PR DESCRIPTION
**Pull request by Nikhil-Mysten**
## Description 

removed the useEffect in favor of useQuery

## Test Plan 

Locally

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [x] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/explorer/src/components/validator-map/index.tsx [hunk 1]
```diff
@@ -85,23 +76,26 @@ export default function ValidatorMap({ minHeight }: Props) {
...
+            `validator-map`,
+            {
+                network: network.toLowerCase(),
+            },
+            {
+                cache: 'no-cache',
+            }
+        ).then((res) => {
+            const data = res as ValidatorMapData[];
+            setValidatorCount(data.length);
```
Comment (Jordan-Mysten) [inline apps/explorer/src/components/validator-map/index.tsx:94]: Don't need separate state for this.

File: apps/explorer/src/components/validator-map/index.tsx [hunk 2]
```diff
@@ -85,23 +76,26 @@ export default function ValidatorMap({ minHeight }: Props) {
...
+            // Some validators will come back as null from the location API
+            return data.filter((validator) => validator);
```
Comment (Jordan-Mysten) [inline apps/explorer/src/components/validator-map/index.tsx:96]: You can use a `selector` for this instead.

File: apps/explorer/src/components/validator-map/index.tsx [hunk 3]
```diff
@@ -85,23 +76,26 @@ export default function ValidatorMap({ minHeight }: Props) {
...
+        isLoading,
+        isError,
+    } = useQuery(['validator-map'], async () =>
+        appsBe(
+            `validator-map`,
+            {
+                network: network.toLowerCase(),
+            },
+            {
+                cache: 'no-cache',
```
Comment (Jordan-Mysten) [inline apps/explorer/src/components/validator-map/index.tsx:90]: Why no-cache?

Comment (Nikhil-Mysten) [reply apps/explorer/src/components/validator-map/index.tsx:90]: left that in by mistake for testing, will remove

File: apps/explorer/src/components/validator-map/index.tsx [hunk 4]
```diff
@@ -85,23 +76,21 @@ export default function ValidatorMap({ minHeight }: Props) {
...
-                // Some validators will come back as null from the API
-                const validatorResponse = data.filter((validator) => validator);
-                setValidatorData(validatorResponse);
-                setIsLoading(false);
-            })
-            .catch((err) => {
-                setIsError(true);
-            });
-    }, [appsBe, network]);
+    const {
```
[the remark this answers is not in the document]
Comment (Nikhil-Mysten) [reply apps/explorer/src/components/validator-map/index.tsx:79]: it's only getting used in the one place, doesn't really work for much else

File: apps/explorer/src/components/validator-map/index.tsx [hunk 5]
```diff
@@ -85,23 +76,21 @@ export default function ValidatorMap({ minHeight }: Props) {
...
+        data: validatorData,
+        isLoading,
+        isError,
+    } = useQuery(
+        ['validator-map'],
+        async () =>
+            appsBe(`validator-map`, {
+                network: network.toLowerCase(),
+            }).then((res) => res as ValidatorMapData[]),
```
Comment (Jordan-Mysten) [inline apps/explorer/src/components/validator-map/index.tsx:88]: I think the `appsBe` function is a generic, so you shouldn't need this `then

File: apps/explorer/src/components/validator-map/index.tsx [hunk 6]
```diff
@@ -85,23 +76,21 @@ export default function ValidatorMap({ minHeight }: Props) {
...
-            });
-    }, [appsBe, network]);
+    const {
+        data: validatorData,
+        isLoading,
+        isError,
+    } = useQuery(
+        ['validator-map'],
+        (): Promise<ValidatorMapData[]> =>
+            appsBe(`validator-map`, {
```
Comment (plam-ml) [inline apps/explorer/src/components/validator-map/index.tsx:86]: don't need the backtick here

File: apps/explorer/src/components/validator-map/index.tsx [hunk 7]
```diff
@@ -85,23 +76,20 @@ export default function ValidatorMap({ minHeight }: Props) {
...
-            .catch((err) => {
-                setIsError(true);
-            });
-    }, [appsBe, network]);
+    const {
+        data: validatorData,
+        isLoading,
+        isError,
+    } = useQuery({
+        queryKey: 'validator-map' as unknown as QueryKey,
```
Comment (Nikhil-Mysten) [inline apps/explorer/src/components/validator-map/index.tsx:84]: getting linting errors for some reason here, so using the object syntax

Comment (Jordan-Mysten) [reply apps/explorer/src/components/validator-map/index.tsx:84]: This should be in an array?

**Review by mamos-mysten — APPROVED** [review]

File: apps/explorer/src/components/validator-map/index.tsx [hunk 8]
```diff
@@ -48,10 +42,7 @@ interface Props {
...
 export default function ValidatorMap({ minHeight }: Props) {
     const [network] = useNetwork();
-    const [validatorData, setValidatorData] = useState<ValidatorMapData[]>([]);
-    const [validatorCount, setValidatorCount] = useState<number>();
-    const [isLoading, setIsLoading] = useState(false);
-    const [isError, setIsError] = useState(false);
+
     const appsBe = useAppsBackend();
 
     const { data: nodeData, isSuccess } = useQuery(['node-map'], async () => {
```
Comment (Jordan-Mysten) [inline apps/explorer/src/components/validator-map/index.tsx:48]: As a further follow-up we still need to migrate off of this

**Pull request merged**